maildir2gmail.py

```diff
@@ -56,12 +59,29 @@ def append(self, filename):
 
         message = email.message_from_string(content)
         timestamp = parsedate(message['date'])
-        if not timestamp:
-            log('Skipping "%s" - no date' % os.path.basename(filename))
+        fileTimestamp = os.path.getctime(filename)
+	if not self.ignore_missing_date:
+            if not timestamp:
+                log('Skipping "%s" - no date (creation time of file: %s)' % (os.path.basename(filename), datetime.datetime.fromtimestamp(fileTimestamp).strftime('%Y-%m-%d %H:%M:%S')))
+                return
+        else:
+            timestamp = os.path.getctime(filename)
```

Comment:
Out of interest: if the message has no "Date" header, how is Gmail sorting things? And does it confuse Gmail, or does everything still work?

Reply:
That's a good question. We should try it out.